docker/api/client.py

```diff
@@ -433,3 +438,73 @@ def get_adapter(self, url):
     @property
     def api_version(self):
         return self._version
+
+
+class NvidiaAPIClient(APIClient):
+    """
+    Version of APIClient that uses the nvidia-docker API to support nvidia GPUs
+
+    Requires `nvidia-docker` to be installed in order to function correctly
```

Comment:
That is up for debate. But this requirement is *only* for the nvidia support. All other use of docker-py will not require this of course. This is captured in the `nvidia_docker_compatible` function. So I'm open to suggestions to changing the logic of that function. Right now it checks three things

1. Is this linux. Nvidia-docker only works on non-virtual machine instances of linux dockers. So this is limited to Linux hosts only
2. Is the nvidia-driver installed. If you aren't using an nvidia card or the nvidia driver (nouveau for example) then there is use for this either
3. Is the `nvidia-docker` executable found. This would indicate the ["plugin"](https://github.com/NVIDIA/nvidia-docker) is installed

The `nvidia-docker` plugin does two things.

1. Create a special `nvidia-docker` mount called `nvidia_driver_{version_number}`. This includes all the .so files and binaries necessary to use nvidia/cuda in a docker. This is not a simple process and is all coded up in go in the [nvidia-docker](https://github.com/NVIDIA/nvidia-docker/blob/3422bca272d64600ca15874d96e4e511b669c7c4/src/nvidia/volumes.go#L104-L173) project. Due to the complexity and superior knowledge nvidia will have and the chance that they will be updating this, I choose to not even consider reproducing this.
2. Mounting the necessary devices and a special nvidia driver library mount into the container.

I'm reproducing the second of these two in python. The first is accomplished by calling `nvidia-docker` on a dummy container to trigger it in `create_nvidia_driver`. This is why I choose to make `nvidia-docker` a requirement for using the `NvidiaAPIClient`.